src/qililab/qprogram/qblox_compiler.py

```diff
@@ -167,7 +178,7 @@ def _append_to_waveforms_of_bus(self, bus: str, waveform_I: Waveform, waveform_Q
         """
 
         def handle_waveform(waveform: Waveform | None, default_length: int = 0):
-            _hash = QbloxCompiler._hash(waveform) if waveform else f"zeros {default_length}"
+            _hash = QbloxCompiler._hash_waveform(waveform) if waveform else f"zeros {default_length}"
 
```

Comment:
At some discussions with the hardware team they pointed out that an envelope / pulse of 0 amplitude should not be treated as an empty pulse (because the electronics might still introduce noise).
I don't know if it would be too difficult to do like in the current qblox compiler and just add a wait when we want to send an empty envelope (I don't know in which context it would make sense to set the default of an empty pulse to amplitude otherwise)

Reply:
Q1ASM `play` instruction expects two waveforms always. If you don't need the second path (for example you are in a flux line bus) you have to set the output mappings accordingly, but this is not QProgram-related.

![image](https://github.com/qilimanjaro-tech/qililab/assets/3442573/c51fe1fe-24a8-4b3d-a972-4b0f4f56df39)